content/how-to/gql/query-mutate-subscribe.md

```diff
@@ -0,0 +1,353 @@
+
+In GraphQL, an _operation_ is a type of request to the server.
+Rhize supports three types of operations:
+
+- **Queries** return data and subsets of data.
+- **Mutations** change the data on the server side.
+- **Subscriptions** notify about data changes in real time.
+
+## Queries
+
+Most queries start with these three verbs, each of which indicates the resources to return.
+- `get` for a single resource
+- `query` for multiple resources
+- `aggregate` for calculations on arrays
+
+### `get` single resource
+
+Queries that start with `get` return one object.
+Common use cases of gets are to explore all data related to a particular object.
+For example, in a custom dashboard, you may use `getDataSource` to make a custom page that reports a specified data source.
+
+
+Typically, the argument specifies the resource by either its human-readable ID (`id`), or its unique address in the database (`iid`).
+
+
+For example, this query gets the `iid`, `_createdBy`, and `versions` for the equipment item `Kitchen_mixer_b_01`:
+
+```graphql
+query mixerCheck {
+  getEquipment(id: "Kitchen_mixer_b_01") {
+      iid
+      _createdBy
+      versions{
+        id
+      }
+  }
+}
+```
+
+### `Query` multiple resources
+
+Queries that start with `query` return an array of objects.
+For example, a custom dashboard may use `queryEquipmentVersion` to create a page that displays all active versions of equipment that are running in a certain {{< abbr "hierarchy scope" >}}.
+
+For example, this query returns the id of all pieces of equipment.
+
+```graphql
+query allEquipment{
+  queryEquipment {
+    id
+  }
+}
+```
+
+
+{{< tabs >}}
+{{% tab "query" %}}
+```graphql
+query allEquipment($filter: EquipmentFilter) {
+g
+  queryEquipment(filter: $filter) {
+    id
+  }
+  }
+}
+```
+{{% /tab %}}
+{{% tab "variables" %}}
+
+```json
+{
+  "EquipmentFilter": {
+    "id": {
+      "regexp": "/KITCHEN_.*|/i"
+    }
+  },
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+Often, its convenient to [filter]({{< relref "call-the-graphql-api#filter" >}}) the items returned by a query.
+This query uses a regular expression in its variables to filter for items that begin with either `Kitchen_` or `Cooling_`, case insensitive:
+
+{{< tabs >}}
+{{% tab "query" %}}
+```graphql
+query getEquipment($filter: EquipmentFilter) {
+  aggregateEquipment(filter: $filter) {
+    count
+  }
+  queryEquipment(filter: $filter) {
+    id
+  }
+  }
+}
+```
+{{% /tab %}}
+{{% tab "variables" %}}
+
+```json
+{
+  "EquipmentFilter": {
+    "id": {
+      "regexp": "/|Kitchen_.*|Cooling_.*/i"
+    }
+  },
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+### `Aggregate` data from multiple resources
+
+Like `query`, operations that start with `aggregate` return sets of arrays.
+
+However, the purpose of aggregate is to returns secondary data computed from the properties of the items.
```

Comment:
```suggestion
However, rather than return items, the purpose of aggregate operations is to returns secondary data computed from the properties of these items.
```